spacy_llm/pipeline/llm.py

```diff
@@ -148,8 +152,19 @@ def _process_docs(self, docs: List[Doc]) -> List[Doc]:
         """
         is_cached = [doc in self._cache for doc in docs]
         noncached_doc_batch = [doc for i, doc in enumerate(docs) if not is_cached[i]]
-        prompts = self._task.generate_prompts(noncached_doc_batch)
-        responses = self._backend(prompts)
+        if len(noncached_doc_batch) < len(docs):
+            logger.debug(
+                "Found %d docs in cache. Processing %d docs not found in cache",
+                len(docs) - len(noncached_doc_batch),
+                len(noncached_doc_batch),
+            )
+
+        prompts = list(self._task.generate_prompts(noncached_doc_batch))
+        responses = list(self._backend(prompts))
+        for prompt, response, doc in zip(prompts, responses, noncached_doc_batch):
+            logger.debug("Generated prompt for doc: %s\n%s", doc.text, prompt)
+            logger.debug("Backend response for doc: %s\n%s", doc.text, response)
```

Comment:
We discussed internally we want to move away from the term `Backend` in user-facing docs and code, as it introduces unnecessary complication. (Stuff will mostly work the same, we'll just operate in terms of "models" instead of "backends" and focus on the model over the backend used to access it).
So let's use "LLM" or "Model" instead of "Backend" here.
```suggestion
            logger.debug("LLM response for doc: %s\n%s", doc.text, response)
```